pyccel/ast/core.py

```diff
@@ -1302,29 +1378,44 @@ def __init__(
         self._variables = variables
         self._body = body
         self._imports = imports
-
         super().__init__()
 
     @property
     def name(self):
+        """ Name of the executable
+        """
         return self._name
 
     @property
     def variables(self):
+        """ Variables contained within the program
+        """
         return self._variables
 
     @property
     def body(self):
+        """ Statements in the program
+        """
         return self._body
 
     @property
     def imports(self):
+        """ Imports imported in the program
+        """
         return self._imports
 
     @property
     def declarations(self):
+        """ Returns the declarations of the variables
+        """
         return [Declare(i.dtype, i) for i in self.variables]
 
+    def remove_import(self, name):
+        """ Remove an import with the given source name from the list
+        of imports
+        """
+        self._imports = tuple(i for i in self.imports if i.source != name)
```

Comment:
if the imports can be changed, why we store them in a tuple can't we use list for easy del and insert?

Reply:
Attribute nodes of `Basic` objects must be other `Basic` objects or tuples ([see here](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/basic.py#L61)). This simplifies things in `pyccel.ast.Basic` (I only need to check `if isinstance(obj, tuple)` rather than `if isinstance(obj, iterable)` or similar). But more importantly it discourages people from modifying these objects without thinking about it (which I apparently did) as we need to update the user nodes/attribute nodes. I have now fixed this: 14e0b1c7

Reply:
It's also not a huge problem as this function is only called once (at least for now, and only when translating to python), so it doesn't matter if this function is not as optimised as it could be

Reply:
hmm, now I see.